_posts/2024-11-28-ranger.md

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title: "The long journey to Apache Ranger"
+author: "Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/images/logos/apache-ranger.png
+---
+
+[Apache Ranger]({{site.baseurl}}/ecosystem/add-on.html#apache-ranger) has
+arrived! With the new [Trino
+466]({{site.baseurl}}/docs/current/release/release-466.html) you all get another
+jam-packed release of Trino awesomeness. One of the goodies is a new plugin for
+access control for your data with Apache Ranger, and it has gone through a long
+story to get here.
+
+<!--more-->
+
+## The beginnings
+
+Apache Ranger has been in use with Trino for a long time - in fact there are
+[early](https://github.com/trinodb/trino/pull/244)
+[rudimentary](https://github.com/trinodb/trino/pull/1069) pull requests from
+2019 that implemented some support. And even before then various hacks existed.
+In 2020 a plugin for PrestoSQL was added to Apache Ranger. Aakash Nand blogged
+about [Integrating Trino and Apache
+Ranger](https://towardsdatascience.com/integrating-trino-and-apache-ranger-b808f6b96ad8)
+in 2021 to adjust for the changes to Trino. Jeff Xu followed up with
+[Integrating Trino and Apache Ranger in a Kerberos secured enterprise
+environment](https://medium.com/@jeff.xu.z/integrating-trino-and-apache-ranger-in-a-kerberos-secured-enterprise-environment-997c95cd10e9)
+in 2022, followed quickly by the addition of the Trino support to the Apache
+Ranger repository.
+
+## Testing and container images
+
+However that was only half of the needed support. The Trino project moves very
+fast with nearly weekly releases, so the best approach is to have the supporting
+plugin in Trino directly so every release includes the relevant updates. [Erik
+Anderson](https://github.com/dprophet) created a more mature plugin that was in
+production use for quite a while for Trino. His [pull request from July
+2022](https://github.com/trinodb/trino/pull/13297) included great background
+reasoning for having the plugin in Trino. One of the issues that Erik solved for
+the Trino project is testing. Trino plugins require the availability of a
+container image for testing whatever integration. Apache Ranger did still not
+ship a container in 2022, but thanks to the lobbying efforts of Erik this
+changed and a container image became available over the months.
```

Comment:
Note that the plugin that we merged does not use a container image for testing. The policies are loaded from an on-disk cache and there is no separate Ranger Admin running in the tests.

Reply:
oh ... do you think we should add that info in this blog post?

Reply:
I don't know. I just mention it because while reading this this seems like relevant information, but in the end those containers are not used for testing here :)

Reply:
Thats good... I worked that info in.